### ISSUE TITLE
Compatibility with NodeBB 1.4.3

### ISSUE THREAD

**Issue by pofider** [NodeBB/nodebb-plugin-category-notifications#7]
I get the following report when the new topic is supposed to be notified through email

```
16/3 12:05:15 [29116] - [31merror[39m:  TypeError: Cannot read property 'cid' of undefined
    at sendTopicEmail (E:\work\temp\NodeBB\node_modules\nodebb-plugin-category-notifications\lib\categoryNotifications.js:123:22)
    at E:\work\temp\NodeBB\node_modules\nodebb-plugin-category-notifications\lib\categoryNotifications.js:71:4
    at E:\work\temp\NodeBB\src\meta\settings.js:14:4
    at handleCallback (E:\work\temp\NodeBB\node_modules\mongodb\lib\utils.js:95:56)
    at E:\work\temp\NodeBB\node_modules\mongodb\lib\collection.js:1405:5
    at handleCallback (E:\work\temp\NodeBB\node_modules\mongodb\lib\utils.js:95:56)
    at E:\work\temp\NodeBB\node_modules\mongodb\lib\cursor.js:674:5
    at handleCallback (E:\work\temp\NodeBB\node_modules\mongodb-core\lib\cursor.js:171:5)
    at nextFunction (E:\work\temp\NodeBB\node_modules\mongodb-core\lib\cursor.js:683:5)
    at E:\work\temp\NodeBB\node_modules\mongodb-core\lib\cursor.js:594:7
    at queryCallback (E:\work\temp\NodeBB\node_modules\mongodb-core\lib\cursor.js:232:18)
    at E:\work\temp\NodeBB\node_modules\mongodb-core\lib\connection\pool.js:455:18
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
```

It seems that data passed to [onTopicPost](https://github.com/NodeBB/nodebb-plugin-category-notifications/blob/master/lib/library.js#L24) are the topic itself:
```js
{"tid":11,"uid":3,"cid":"2","mainPid":12,"title":"fe fwf efwe few few fewf ewf","slug":"11/fe-fwf-efwe-few-few-fewf-ewf","timestamp":1489662849429,"lastposttime":1489662849429,"postcount":1,"viewcount":0,"locked":false,"deleted":false,"pinned":false,"titleRaw":"fe fwf efwe few few fewf ewf","timestampISO":"2017-03-16T11:14:09.429Z","lastposttimeISO":"2017-03-16T11:14:09.429Z","category":{"cid":2,"name":"General Discussion","icon":"fa-comments-o","bgColor":"#59b3d0","color":"#fff","slug":"2/general-discussion","disabled":false,"image":null},"user":{"username":"pofider","userslug":"pofider","picture":"","fullname":"","signature":"","reputation":0,"postcount":8,"banned":0,"status":"online","uid":3,"icon:text":"P","icon:bgColor":"#673ab7"},"tags":[],"isOwner":true,"ignored":false,"unread":false,"bookmark":null,"unreplied":1,"icons":[],"mainPost":{"pid":12,"uid":3,"tid":11,"content":"<p>ewf efewf ewf ewfew fewf wfwe fwf ew fe fewf ew fwfw</p>\n","timestamp":1489662849429,"deleted":0,"cid":"2","isMain":true,"user":{"username":"pofider","userslug":"pofider","lastonline":1489662696601,"picture":"","fullname":"","signature":"","reputation":0,"postcount":8,"banned":false,"status":"online","uid":3,"groupTitle":null,"icon:text":"P","icon:bgColor":"#673ab7","lastonlineISO":"2017-03-16T11:11:36.601Z","custom_profile_info":[]},"topic":{"tid":11,"title":"fe fwf efwe few few fewf ewf","slug":"11/fe-fwf-efwe-few-few-fewf-ewf","cid":"2","postcount":1,"mainPid":12},"index":0,"votes":0,"bookmarked":false,"display_edit_tools":true,"display_delete_tools":true,"display_moderator_tools":true,"display_move_tools":true,"selfPost":false,"timestampISO":"2017-03-16T11:14:09.429Z"}}
```

It seems to work when changing `onTopicPost` handler like this
```js
library.onTopicPost = function(data) {	
	categoryNotification.onTopicPost(data.topic || data);
};
```

Thoughts? Do you want PR for this change?

**Comment by barisusakli** [comment]
What version are you using, I think this is fixed in https://github.com/NodeBB/nodebb-plugin-category-notifications/commit/c3e7cc64183d6c6c66b2cffa15f56d8cdff378d4

**Comment by pofider** [comment]
> What version are you using

3

>  I think this is fixed in c3e7cc6

That commit actually breaks it, based on what I wrote

The fork [pofider/nodebb-plugin-category-notifications](https://github.com/pofider/nodebb-plugin-category-notifications) fix the issue for me.

**Comment by barisusakli** [comment]
2.0.3 should work on nodebb 1.4.3
3.0.0 should work on nodebb 1.5.0 and up

**Comment by pofider** [comment]
Hm, I would expect that nodebb marked with tag latest would work with plugin marked with the latest.
I don't see 1.5.0 nodebb even released yet - https://github.com/NodeBB/NodeBB/tags

In every case, I just wanted to help the others having the same problem. Compatibility of plugins is challenge in every system like this.  Closing as this is solved for me in.

Thank you for your time.

**Comment by barisusakli** [comment]
Yeah we have the nbb compatibility flag in https://github.com/NodeBB/nodebb-plugin-category-notifications/blob/master/package.json#L25-L27

The acp should not show version 3.0.0 for nodebb 1.4.x, once nodebb 1.5.0 is released it will show 3.0.0 of this plugin. But if you install the latest of this plugin from cli then it wouldn't work for 1.4.x

**Comment by pofider** [comment]
Oh yes, I did install through npm. That is it. Thank you for clarification.
